### PR TITLE
Added more test cases for FileSystem::GetNormalizedPath

### PR DESCRIPTION
**Pull request by dalpra-hcd**

### CODE REVIEW

File: test/engine/System/FileSystem/TestFileSystem.cpp [hunk 1]
```diff
@@ -133,14 +133,123 @@ TEST_CASE("GetDirectory")
...
+	CHECK_NORM_PATH("/foo/../bar", "/bar");
+	CHECK_NORM_PATH("/foo/./bar", "/foo/bar");
+}
+
+TEST_CASE("GetNormalizedPath - trailing slashes") 
+{
+	CHECK_NORM_PATH("foo/bar/", "foo/bar");
+	CHECK_NORM_PATH("foo/bar//", "foo/bar");
+	CHECK_NORM_PATH("./foo/", "./foo");
+}
```
Comment (sprunk) [inline test/engine/System/FileSystem/TestFileSystem.cpp:212]: Could have a testcase for windows style slash too.

File: test/engine/System/FileSystem/TestFileSystem.cpp [hunk 2]
```diff
@@ -133,14 +133,123 @@ TEST_CASE("GetDirectory")
...
+
+TEST_CASE("GetNormalizedPath - current directory") 
+{
+	CHECK_NORM_PATH("./foo/bar", "./foo/bar");
+	CHECK_NORM_PATH(".\\foo\\bar", "./foo/bar");
+	CHECK_NORM_PATH("foo/./bar", "foo/bar");
+	CHECK_NORM_PATH("foo/.", "foo");
+	CHECK_NORM_PATH(".", ".");
+	CHECK_NORM_PATH("./", ".");
+	CHECK_NORM_PATH("./.", ".");
```
Comment (sprunk) [inline test/engine/System/FileSystem/TestFileSystem.cpp:163]: could have a test with `.` at the end that doesn't resolve to `.` (e.g. `"foo/." -> "foo"`)

**Comment by dalpra-hcd** [comment]
@lhog @p2004a Please have a look as well.
I tried adding more unit tests for edge cases, but I can't judge what is the correct assumption or not.

File: test/engine/System/FileSystem/TestFileSystem.cpp (hunk 1 above)
Comment (dalpra-hcd) [reply test/engine/System/FileSystem/TestFileSystem.cpp:212]: Could you give some examples as well? It's been a long time since I worked with Windows.

Comment (sprunk) [reply test/engine/System/FileSystem/TestFileSystem.cpp:212]: e.g.
```
	CHECK_NORM_PATH("foo\\bar\\", "foo/bar");
	CHECK_NORM_PATH("foo\\bar\\\\", "foo/bar");
	CHECK_NORM_PATH(".\\foo\\", "./foo");
	CHECK_NORM_PATH("C:\\foo\\", "C:/foo");
```

**Comment by lhog** [comment]
Perhaps rebase it on top of https://github.com/beyond-all-reason/RecoilEngine/tree/fixTests-v2 as it changed the `GetNormalizedPath` implementation a bit

**Comment by dalpra-hcd** [comment]
> Perhaps rebase it on top of https://github.com/beyond-all-reason/RecoilEngine/tree/fixTests-v2 as it changed the `GetNormalizedPath` implementation a bit

I prefer to get the tests first and then fix later.

File: test/engine/System/FileSystem/TestFileSystem.cpp (hunk 2 above)
Comment (dalpra-hcd) [reply test/engine/System/FileSystem/TestFileSystem.cpp:163]: > could have a test with `.` at the end that doesn't resolve to `.` (e.g. `"foo/." -> "foo"`)

Sorry I see one already on line 160?
